tempodb/encoding/vparquet4/schema_test.go

```diff
@@ -919,10 +923,86 @@ func BenchmarkExtendReuseSlice(b *testing.B) {
 	}
 }
 
-// assertEqualEquateEmpty asserts similar to assert.Equal but treats empty / nil slices and maps as equal
-func assertEqualEquateEmpty(t *testing.T, expected, actual interface{}, messages ...interface{}) {
+var benchmarkTrace *Trace
+
+func BenchmarkTraceToParquet(b *testing.B) {
+	var (
+		traceID       = test.ValidTraceID(nil)
+		traces        = make([]*tempopb.Trace, 0, 1_000)
+		parquetTraces = make([]*Trace, 11)
+	)
+
+	for range 1_000 {
+		nb := 20 + rand.Intn(5)
+		id := test.ValidTraceID(nil)
+		traces = append(traces, test.AddDedicatedAttributes(test.MakeTrace(nb, test.ValidTraceID(id))))
+	}
+
+	b.ResetTimer()
+	b.ReportAllocs()
+
+	for i := range b.N {
```

Comment:
go 1.24 has the cool new `b.Loop()`!

you're welcome to use or ignore, just learned about this myself.

Reply:
> just learned about this myself

Good suggestion! Definitely looking forward to using `b.Loop()` more often instead of the old `for i := 0; i < b.N; i++`. However, in this case I'm actually using the `i` to select one of the pre-allocated traces